presto-hive/src/main/java/io/prestosql/plugin/hive/DropStatsProcedure.java

```diff
@@ -0,0 +1,169 @@
+        this.metastore = requireNonNull(metastore, "metastore is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "drop_stats",
+                ImmutableList.of(
+                        new Argument("schema_name", VARCHAR),
+                        new Argument("table_name", VARCHAR),
+                        new Argument("partition_values", new ArrayType(new ArrayType(VARCHAR)))),
+                DROP_STATS.bindTo(this));
+    }
+
+    public void dropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doDropStats(session, schema, table, partitionValues);
+        }
+    }
+
+    private void doDropStats(ConnectorSession session, String schema, String table, List<?> partitionValues)
+    {
+        TransactionalMetadata hiveMetadata = hiveMetadataFactory.get();
+        HiveTableHandle handle = (HiveTableHandle) hiveMetadata.getTableHandle(session, new SchemaTableName(schema, table));
+        Map<String, ColumnHandle> columns = hiveMetadata.getColumnHandles(session, handle);
+        List<String> partitionColumns = columns.values().stream()
+                .map(HiveColumnHandle.class::cast)
+                .filter(HiveColumnHandle::isPartitionKey)
+                .map(HiveColumnHandle::getName)
+                .collect(toImmutableList());
+
+        if (partitionValues != null) {
+            // drop stats for specified partitions
+            List<List<String>> partitionStringValues = partitionValues.stream()
+                    .map(DropStatsProcedure::validateParameterType)
+                    .collect(ImmutableList.toImmutableList());
+            validatePartitions(partitionStringValues, partitionColumns);
+
+            partitionStringValues.forEach(values -> metastore.updatePartitionStatistics(
+                    new HiveIdentity(session.getIdentity()),
+                    schema,
+                    table,
+                    makePartName(partitionColumns, values),
+                    stats -> PartitionStatistics.empty()));
+        }
+        else {
+            // no partition specified, so drop stats for the entire table
+            if (partitionColumns.isEmpty()) {
+                // for non-partitioned tables, just wipe table stats
+                metastore.updateTableStatistics(
+                        new HiveIdentity(session.getIdentity()),
+                        schema,
+                        table,
+                        stats -> PartitionStatistics.empty());
+            }
+            else {
+                // the table is partitioned; remove stats for every partition
+                metastore.getPartitionNames(new HiveIdentity(session.getIdentity()), handle.getSchemaName(), handle.getTableName())
```

Comment:
nit: We can suppress NPE warning by
```suggestion
                metastore.getPartitionNames(new HiveIdentity(session.getIdentity()), schema, table)
```